src/lib/features/change-request-access-service/createChangeRequestAccessReadModel.ts

```diff
@@ -0,0 +1,19 @@
+import { Db, IUnleashConfig } from 'lib/server-impl';
+import { ChangeRequestAccessReadModel } from './sql-change-request-access-read-model';
+import { createAccessService } from '../access/createAccessService';
+import { FakeChangeRequestAccessReadModel } from './fake-change-request-access-read-model';
+import { IChangeRequestAccessReadModel } from './change-request-access-read-model';
+
+export const createChangeRequestAccessReadModel = (
+    db: Db,
+    config: IUnleashConfig,
+): IChangeRequestAccessReadModel => {
+    const accessService = createAccessService(db, config);
+
+    return new ChangeRequestAccessReadModel(db, accessService);
+};
+
+export const createFakeChangeRequestAccessService =
+    (): IChangeRequestAccessReadModel => {
+        return new FakeChangeRequestAccessReadModel(true);
+    };
```

Comment:
I noticed this pattern of real object + fake object available in the same file. I assume fake is usually for testing, would it make sense to have it in a `.test.ts` file? As usual, this could be just my java-centric mind, but also having this in "production accessible code" makes me fear that by accident someone forgets to remove the fake implementation before going to production... I have some ideas, but first I'd like to understand the context of this design decision

Reply:
I'm curious about the use cases where you'd use this. If this is a new feature behind a flag, you can easily write the fake code in the real implementation which later can evolve to be the real implementation. That way you only deal with "the implementation" and not a fake vs a real one. 

Other scenarios I think can easily be dealt with Feature Toggles which would give you control at runtime rather than at boot time and you can still have that in-memory behavior if you want.

The risk I see in having these fakes mixed with production code is that you may end up running something that differs from what customers run in production, and this is one thing I don't like to sacrifice.

Reply:
What about a common closure principle then? Things that change together should be kept together. When I change a contract for real and fake impl I want to do it inside one directory. I don't want to do [shotgun surgery](https://refactoring.guru/smells/shotgun-surgery). I'd like all my feature changes to be done in a feature directory. If I delete a feature I want to delete one directory. Etc.

Reply:
Actually if you look at the tests they are also in the same directory. This is where Java and JS/TS nodes diverged. I see JS/TS to follow common closure principle to a greater extent. So when we say feature-oriented architecture then everything about this feature is in a feature directory.

Reply:
Yes, but one thing is having tests in the same directory, but not being able to refer to a test class/object from production code because somehow `.test.ts` files are not accessible (I don't know much about the implementation details of the typescript compiler here). But having test code being accessible from production code seems a bit risky because the boundary becomes blurry.